### ISSUE TITLE
Compilation failed in centos8 system

### ISSUE THREAD

**Issue by leaningchen** [CppMicroServices/CppMicroServices#707]
-- Detected Linker: /usr/bin/ld
Framework version is 3.7.2
HttpService version is 0.1.0
WebConsole version is 0.1.0
ShellService version is 0.1.0
LogServiceImpl version is 1.0.0
DeclarativeServices version is 1.3.1
ConfigurationAdmin version is 1.3.0
-- Configuring done
CMake Error at tools/rc/CMakeLists.txt:13 (add_executable):
  Target "usResourceCompiler" links to target "nowide::nowide" but the target
  was not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?

-- Generating done
CMake Generate step failed.  Build files cannot be regenerated correctly.

the environmental information:
root@localhost:__distributes# cat /etc/redhat-release
CentOS Stream release 8
root@localhost:__distributes# gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/libexec/gcc/x86_64-redhat-linux/8/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none
OFFLOAD_TARGET_DEFAULT=1
Target: x86_64-redhat-linux
Configured with: ../configure --enable-bootstrap --enable-languages=c,c++,fortran,lto --prefix=/usr --mandir=
/usr/share/man --infodir=/usr/share/info --with-bugurl=http://bugzilla.redhat.com/bugzilla --enable-shared --
enable-threads=posix --enable-checking=release --enable-multilib --with-system-zlib --enable-__cxa_atexit --d
isable-libunwind-exceptions --enable-gnu-unique-object --enable-linker-build-id --with-gcc-major-version-only
 --with-linker-hash-style=gnu --enable-plugin --enable-initfini-array --with-isl --disable-libmpx --enable-of
fload-targets=nvptx-none --without-cuda-driver --enable-gnu-indirect-function --enable-cet --with-tune=generi
c --with-arch_32=x86-64 --build=x86_64-redhat-linux
Thread model: posix
gcc version 8.5.0 20210514 (Red Hat 8.5.0-13) (GCC)
root@localhost:__distributes# cmake --version
cmake version 3.20.2

CMake suite maintained and supported by Kitware (kitware.com/cmake).
root@localhost:__distributes#

**Comment by leaningchen** [comment]
[ 10%] Building CXX object tools/rc/CMakeFiles/usResourceCompiler.dir/ResourceCompiler.cpp.o
/root/workspace/sources/others/CppMicroServices/tools/rc/ResourceCompiler.cpp:38:10: fatal error: nowide/args
.hpp: No such file or directory
 #include <nowide/args.hpp>
          ^~~~~~~~~~~~~~~~~
compilation terminated.
gmake[2]: *** [tools/rc/CMakeFiles/usResourceCompiler.dir/build.make:76: tools/rc/CMakeFiles/usResourceCompil
er.dir/ResourceCompiler.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:538: tools/rc/CMakeFiles/usResourceCompiler.dir/all] Error 2
gmake: *** [Makefile:136: all] Error 2

**Comment by leaningchen** [comment]
> Yes, all the codes have been downloaded successfully

**Comment by leaningchen** [comment]
@achristoforides Yes, all the codes have been downloaded successfully

**Comment by leaningchen** [comment]
@achristoforides all the codes have been downloaded successfully
root@localhost:CppMicroServices# ll third_party/boost/
total 16
drwxr-xr-x  4 root root 4096 Jul 20 04:27 ./
drwxr-xr-x 12 root root 4096 Jul 20 04:27 ../
drwxr-xr-x  3 root root 4096 Jul 20 04:27 include/
drwxr-xr-x 14 root root 4096 Jul 20 23:12 nowide/
root@localhost:CppMicroServices# ll third_party/boost/nowide/
total 124
drwxr-xr-x 14 root root 4096 Jul 20 23:12 ./
drwxr-xr-x  4 root root 4096 Jul 20 04:27 ../
-rw-r--r--  1 root root 7480 Jul 20 23:12 .appveyor.yml
-rw-r--r--  1 root root 8669 Jul 20 23:12 .azure-pipelines.yml
-rw-r--r--  1 root root 4108 Jul 20 23:12 .clang-format
-rw-r--r--  1 root root  204 Jul 20 23:12 .codecov.yml
drwxr-xr-x  8 root root 4096 Jul 20 23:13 .git/
-rw-r--r--  1 root root 3917 Jul 20 23:12 .gitattributes
drwxr-xr-x  3 root root 4096 Jul 20 23:12 .github/
-rw-r--r--  1 root root  107 Jul 20 23:12 .gitignore
-rw-r--r--  1 root root 3553 Jul 20 23:12 CMakeLists.txt
-rw-r--r--  1 root root  376 Jul 20 23:12 Config.cmake.in
-rw-r--r--  1 root root 1338 Jul 20 23:12 LICENSE
-rw-r--r--  1 root root 6741 Jul 20 23:12 README.md
drwxr-xr-x  2 root root 4096 Jul 20 23:12 build/
drwxr-xr-x  2 root root 4096 Jul 20 23:12 cmake/
drwxr-xr-x  2 root root 4096 Jul 20 23:12 config/
drwxr-xr-x  2 root root 4096 Jul 20 23:12 doc/
drwxr-xr-x  3 root root 4096 Jul 20 23:12 include/
-rw-r--r--  1 root root  627 Jul 20 23:12 index.html
drwxr-xr-x  2 root root 4096 Jul 20 23:12 meta/
drwxr-xr-x  2 root root 4096 Jul 20 23:12 src/
drwxr-xr-x  2 root root 4096 Jul 20 23:12 standalone/
drwxr-xr-x  3 root root 4096 Jul 20 23:12 test/
drwxr-xr-x  2 root root 4096 Jul 20 23:12 tools/
-rw-r--r--  1 root root  125 Jul 20 23:12 ubsan-blacklist
root@localhost:CppMicroServices#

2 - cmake cammandline
root@localhost:CppMicroServices# pwd
/root/workspace/sources/others/CppMicroServices
root@localhost:CppMicroServices# cd __distributes/
root@localhost:__distributes# ll
total 8
drwxr-xr-x  2 root root 4096 Jul 20 23:13 ./
drwxr-xr-x 16 root root 4096 Jul 20 23:09 ../
root@localhost:__distributes# cmake ..
-- The C compiler identification is GNU 8.5.0
-- The CXX compiler identification is GNU 8.5.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Detected Linker: /usr/bin/ld
-- Setting build type to 'Debug' as none was specified.
-- Performing Test HAS_CXX_FLAG__fstack_protector_all
-- Performing Test HAS_CXX_FLAG__fstack_protector_all - Success
-- Performing Test HAS_CXX_FLAG__Werror
-- Performing Test HAS_CXX_FLAG__Werror - Success
-- Performing Test HAS_CXX_FLAG__Wall
-- Performing Test HAS_CXX_FLAG__Wall - Success
-- Performing Test HAS_CXX_FLAG__Wextra
-- Performing Test HAS_CXX_FLAG__Wextra - Success
-- Performing Test HAS_CXX_FLAG__Wpointer_arith
-- Performing Test HAS_CXX_FLAG__Wpointer_arith - Success
-- Performing Test HAS_CXX_FLAG__Winvalid_pch
-- Performing Test HAS_CXX_FLAG__Winvalid_pch - Success
-- Performing Test HAS_CXX_FLAG__Wcast_align
-- Performing Test HAS_CXX_FLAG__Wcast_align - Success
-- Performing Test HAS_CXX_FLAG__Wwrite_strings
-- Performing Test HAS_CXX_FLAG__Wwrite_strings - Success
-- Performing Test HAS_CXX_FLAG__Woverloaded_virtual
-- Performing Test HAS_CXX_FLAG__Woverloaded_virtual - Success
-- Performing Test HAS_CXX_FLAG__Wnon_virtual_dtor
-- Performing Test HAS_CXX_FLAG__Wnon_virtual_dtor - Success
-- Performing Test HAS_CXX_FLAG__Wold_style_cast
-- Performing Test HAS_CXX_FLAG__Wold_style_cast - Success
-- Performing Test HAS_CXX_FLAG__Wstrict_null_sentinel
-- Performing Test HAS_CXX_FLAG__Wstrict_null_sentinel - Success
-- Performing Test HAS_CXX_FLAG__Wsign_promo
-- Performing Test HAS_CXX_FLAG__Wsign_promo - Success
-- Performing Test HAS_CXX_FLAG__fdiagnostics_show_option
-- Performing Test HAS_CXX_FLAG__fdiagnostics_show_option - Success
-- Performing Test HAS_CXX_FLAG__fvisibility_hidden_fvisibility_inlines_hidden
-- Performing Test HAS_CXX_FLAG__fvisibility_hidden_fvisibility_inlines_hidden - Success
-- Performing Test HAS_CXX_FLAG__O1_D_FORTIFY_SOURCE_2
-- Performing Test HAS_CXX_FLAG__O1_D_FORTIFY_SOURCE_2 - Success
-- Performing Test HAS_CXX_FLAG__O1_fsanitize_address_fno_omit_frame_pointer
-- Performing Test HAS_CXX_FLAG__O1_fsanitize_address_fno_omit_frame_pointer - Failed
-- Performing Test HAS_CXX_FLAG__O1_fsanitize_thread
-- Performing Test HAS_CXX_FLAG__O1_fsanitize_thread - Failed
-- Performing Test HAS_CXX_FLAG__fsanitize_memory_fsanitize_memory_track_origins_2_fPIE_fno_omit_frame_pointe
r
-- Performing Test HAS_CXX_FLAG__fsanitize_memory_fsanitize_memory_track_origins_2_fPIE_fno_omit_frame_pointe
r - Failed
-- Performing Test HAS_CXX_FLAG__fsanitize_undefined_fno_sanitize_recover_all_fno_omit_frame_pointer
-- Performing Test HAS_CXX_FLAG__fsanitize_undefined_fno_sanitize_recover_all_fno_omit_frame_pointer - Failed
-- Performing Test HAS_CXX_FLAG__fprofile_arcs_ftest_coverage
-- Performing Test HAS_CXX_FLAG__fprofile_arcs_ftest_coverage - Success
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
-- Found Threads: TRUE
-- Performing Test BOOST_NOWIDE_HAS_LFS
-- Performing Test BOOST_NOWIDE_HAS_LFS - Success
-- Performing Test BOOST_NOWIDE_HAS_INIT_PRIORITY
-- Performing Test BOOST_NOWIDE_HAS_INIT_PRIORITY - Success
-- Found Boost: /usr/include (found suitable version "1.66.0", minimum required is "1.56")
-- Performing Test HAS_CXX_FLAG__Wl__no_undefined
-- Performing Test HAS_CXX_FLAG__Wl__no_undefined - Success
Checking for CppMicroServices resource linking capability...yes
-- Looking for C++ include cxxabi.h
-- Looking for C++ include cxxabi.h - found
Framework version is 3.7.2
HttpService version is 0.1.0
WebConsole version is 0.1.0
-- Performing Test HAS_CXX_FLAG__Wno_switch
-- Performing Test HAS_CXX_FLAG__Wno_switch - Success
ShellService version is 0.1.0
LogServiceImpl version is 1.0.0
DeclarativeServices version is 1.4.0
ConfigurationAdmin version is 1.3.1
-- Found CppMicroServices: /root/workspace/sources/others/CppMicroServices/__distributes/CppMicroServicesConf
ig.cmake (found version "3.7.2") found components: ShellService
-- Found Doxygen: /usr/local/bin/doxygen (found version "1.9.5 (3e0e97c5e76aec1e4c9a0f6c5a6170c7a579b076*)")
found components: doxygen missing components: dot
-- Configuring done
CMake Error at tools/rc/CMakeLists.txt:13 (add_executable):
  Target "usResourceCompiler" links to target "nowide::nowide" but the target
  was not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


-- Generating done
CMake Generate step failed.  Build files cannot be regenerated correctly.
root@localhost:__distributes#

**Comment by leaningchen** [comment]
@achristoforides cmake build
root@localhost:__distributes# cmake --build .
[  1%] Building CXX object third_party/absl/CMakeFiles/absl.dir/absl/base/log_severity.cc.o
[  1%] Building CXX object third_party/absl/CMakeFiles/absl.dir/absl/base/internal/raw_logging.cc.o
[  2%] Building CXX object third_party/absl/CMakeFiles/absl.dir/absl/base/internal/throw_delegate.cc.o
[  2%] Building CXX object third_party/absl/CMakeFiles/absl.dir/absl/numeric/int128.cc.o
[  3%] Building CXX object third_party/absl/CMakeFiles/absl.dir/absl/strings/ascii.cc.o
[  3%] Building CXX object third_party/absl/CMakeFiles/absl.dir/absl/strings/charconv.cc.o
[  4%] Building CXX object third_party/absl/CMakeFiles/absl.dir/absl/strings/match.cc.o
[  4%] Building CXX object third_party/absl/CMakeFiles/absl.dir/absl/strings/numbers.cc.o
[  5%] Building CXX object third_party/absl/CMakeFiles/absl.dir/absl/strings/str_cat.cc.o
[  5%] Building CXX object third_party/absl/CMakeFiles/absl.dir/absl/strings/string_view.cc.o
[  6%] Building CXX object third_party/absl/CMakeFiles/absl.dir/absl/strings/internal/charconv_bigint.cc.o
[  6%] Building CXX object third_party/absl/CMakeFiles/absl.dir/absl/strings/internal/charconv_parse.cc.o
[  7%] Building CXX object third_party/absl/CMakeFiles/absl.dir/absl/strings/internal/memutil.cc.o
[  7%] Built target absl
[  7%] Building CXX object third_party/boost/nowide/CMakeFiles/boost_nowide.dir/src/console_buffer.cpp.o
[  8%] Building CXX object third_party/boost/nowide/CMakeFiles/boost_nowide.dir/src/cstdio.cpp.o
[  8%] Building CXX object third_party/boost/nowide/CMakeFiles/boost_nowide.dir/src/cstdlib.cpp.o
[  9%] Building CXX object third_party/boost/nowide/CMakeFiles/boost_nowide.dir/src/filebuf.cpp.o
[  9%] Building CXX object third_party/boost/nowide/CMakeFiles/boost_nowide.dir/src/iostream.cpp.o
[ 10%] Building CXX object third_party/boost/nowide/CMakeFiles/boost_nowide.dir/src/stat.cpp.o
[ 10%] Linking CXX static library ../../../lib/libboost_nowided.a
[ 10%] Built target boost_nowide
[ 10%] Building CXX object tools/rc/CMakeFiles/usResourceCompiler.dir/ResourceCompiler.cpp.o
/root/workspace/sources/others/CppMicroServices/tools/rc/ResourceCompiler.cpp:38:10: fatal error: nowide/args
.hpp: No such file or directory
 #include <nowide/args.hpp>
          ^~~~~~~~~~~~~~~~~
compilation terminated.
gmake[2]: *** [tools/rc/CMakeFiles/usResourceCompiler.dir/build.make:76: tools/rc/CMakeFiles/usResourceCompil
er.dir/ResourceCompiler.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:538: tools/rc/CMakeFiles/usResourceCompiler.dir/all] Error 2
gmake: *** [Makefile:136: all] Error 2
root@localhost:__distributes#

**Comment by leaningchen** [comment]
@achristoforides 
![无标题](https://user-images.githubusercontent.com/63138022/180124098-82372dde-bcc9-46f5-af0c-acc1865d20f0.png)

**Comment by achristoforides** [comment]
@leaningchen I tried your reproduction steps on a Debian 11 machine.

I ran the following:
```
(CppMicroServices/build)$ cmake ..
(CppMicroServices/build)$ cmake --build .
```

I noticed that the name of the resulting static library for nowide on your machine is `boost_nowided.a` while on mine (and in all of our CI builds) is `libnowided.a`.

Additionally, the name of the nowide target in your output is `boost_nowide` while mine is `nowide`.

Please see my output for reference:
```
Scanning dependencies of target nowide
[  6%] Building CXX object third_party/boost/nowide/CMakeFiles/nowide.dir/src/cstdio.cpp.o
[  7%] Building CXX object third_party/boost/nowide/CMakeFiles/nowide.dir/src/cstdlib.cpp.o
[  7%] Building CXX object third_party/boost/nowide/CMakeFiles/nowide.dir/src/iostream.cpp.o
[  8%] Linking CXX static library ../../../lib/libnowided.a
[  8%] Built target nowide
Scanning dependencies of target usResourceCompiler
[  9%] Building CXX object tools/rc/CMakeFiles/usResourceCompiler.dir/ResourceCompiler.cpp.o
[  9%] Building C object tools/rc/CMakeFiles/usResourceCompiler.dir/__/__/third_party/miniz.c.o
CppMicroServices/third_party/miniz.c:3181:9: note: ??#pragma message: Using fopen, ftello, fseeko, stat() etc. path for file I/O - this path may not support large files.??
 3181 | #pragma message("Using fopen, ftello, fseeko, stat() etc. path for file I/O - this path may not support large files.")
      |         ^~~~~~~
[ 10%] Building CXX object tools/rc/CMakeFiles/usResourceCompiler.dir/__/__/third_party/jsoncpp.cpp.o
[ 10%] Linking CXX executable ../../bin/usResourceCompiler3
[ 10%] Built target usResourceCompiler
```

Can you `cd` into the `CppMicroServices/third_party/boost/nowide` directory and run `git status` and `git log` to confirm which branch/commit your version of the submodule is on?

**Comment by leaningchen** [comment]
@achristoforides Oh, I see. Is that really not the right branch, I was careless! Thank you very much for your help!

**Comment by achristoforides** [comment]
@leaningchen Glad I could help!

I'm curious though... we added nowide as a git submodule and the branch/commit we pinned the submodule to is https://github.com/boostorg/nowide/tree/02f40f0b5f5686627fcddae93ff88ca399db4766 on their `standalone` branch.

When cloning CppMicroServices via `git clone --recursive https://github.com/CppMicroServices/CppMicroServices`, was the pinned commit of the nowide submodule incorrect? Or did you manually clone nowide yourself in the `CppMicroServices/third_party/boost` directory?
